addons/sale_management/models/sale_order.py

```diff
@@ -108,10 +108,12 @@ def _onchange_sale_order_template_id(self):
         sale_order_template = self.sale_order_template_id.with_context(lang=self.partner_id.lang)
 
         order_lines_data = [fields.Command.clear()]
-        order_lines_data += [
-            fields.Command.create(line._prepare_order_line_values())
-            for line in sale_order_template.sale_order_template_line_ids
-        ]
+        for line in sale_order_template.sale_order_template_line_ids:
+            order_line_values = line._prepare_order_line_values()
```

Comment:
Can't you handle this directly in `_prepare_order_line_values` ?

Reply:
Yes, we can. :+1: